Calibration/TkAlCaRecoProducers/plugins/NearbyPixelClustersProducer.cc

```diff
@@ -0,0 +1,440 @@
+      const std::vector<LocalPoint>& vLocalPos);
+
+  const std::vector<edmNew::DetSet<SiPixelCluster>::const_iterator> findAllNearbyClusters(
+      const SiPixelClusterCollectionNew& clusterSet, const uint32_t rawId, const std::vector<LocalPoint>& vLocalPos);
+
+  TrajectoryStateOnSurface getTrajectoryStateOnSurface(const TrajectoryMeasurement& measurement);
+  bool detidIsOnPixel(const DetId& detid);
+
+  // ----------member data ---------------------------
+
+  // switches
+  const bool throwBadComponents_;
+  const bool dumpWholeDetId_;
+
+  // esTokens
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomEsToken_;
+  const edm::ESGetToken<PixelClusterParameterEstimator, TkPixelCPERecord> pixelCPEEsToken_;
+  const edm::ESGetToken<SiPixelQuality, SiPixelQualityFromDbRcd> badModuleToken_;
+
+  // edToken
+  edm::EDGetTokenT<SiPixelClusterCollectionNew> clustersToken_;
+  edm::EDGetTokenT<TrajTrackAssociationCollection> trajTrackCollectionToken_;
+
+  // putToken
+  edm::EDPutTokenT<SiPixelClusterCollectionNew> clusterPutToken_;
+
+  // event setup
+  const TrackerGeometry* trackerGeometry_;
+  const PixelClusterParameterEstimator* pixelCPE_;
+};
+
+//
+// constructors and destructor
+//
+NearbyPixelClustersProducer::NearbyPixelClustersProducer(const edm::ParameterSet& iConfig)
+    : throwBadComponents_(iConfig.getParameter<bool>("throwBadComponents")),
+      dumpWholeDetId_(iConfig.getParameter<bool>("dumpWholeDetIds")),
+      geomEsToken_(esConsumes()),
+      pixelCPEEsToken_(esConsumes(edm::ESInputTag("", "PixelCPEGeneric"))),
+      badModuleToken_(esConsumes()),
+      clustersToken_(consumes<SiPixelClusterCollectionNew>(iConfig.getParameter<edm::InputTag>("clusterCollection"))),
+      trajTrackCollectionToken_(
+          consumes<TrajTrackAssociationCollection>(iConfig.getParameter<edm::InputTag>("trajectoryInput"))),
+      clusterPutToken_(produces<SiPixelClusterCollectionNew>()) {
+  if (dumpWholeDetId_) {
+    edm::LogPrint("NearbyPixelClustersProducer") << "WARNING: selected to dump the whole DetId's worth of clusters.\n "
+                                                    "This will have consequences on the event size!"
+                                                 << std::endl;
+  }
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void NearbyPixelClustersProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  auto outputClusters = std::make_unique<SiPixelClusterCollectionNew>();
+
+  // get the Tracker geometry from event setup
+  trackerGeometry_ = &iSetup.getData(geomEsToken_);
+
+  // get the Pixel CPE from event setup
+  pixelCPE_ = &iSetup.getData(pixelCPEEsToken_);
+
+  const auto& SiPixelBadModule_ = &iSetup.getData(badModuleToken_);
+
+  // get cluster collection
+  edm::Handle<SiPixelClusterCollectionNew> clusterCollectionHandle;
+  iEvent.getByToken(clustersToken_, clusterCollectionHandle);
```

Comment:
From Clang Tidy checks, a warning is due to direct call of function getByToken(EDGetTokenT<>&, Handle<>&) is deprecated. Could you modify it according to the suggested way: replace here with clusterCollectionHandle = iEvent.getHandle(clustersToken_)?

Similar ones on L. 154 and in NearbyPixelClustersAnalyzer.cc on L. 146, 156 and 169.

Reply:
Will do, but out of curiosity, is this already enforced? I don't remember having seen it announced formally.

Reply:
> From Clang Tidy checks, a warning is due to direct call of function getByToken(EDGetTokenT<>&, Handle<>&) is deprecated

That check can be interpreted as "a mild suggestion" for now. When enforced it will be announced, and we'll try an automated migration (as in this case it would be possible).

Reply:
fixed at 860313f

Reply:
Thank you very much!